controller_manager/src/controller_manager_parameters.yaml

```
@@ -29,6 +29,11 @@ controller_manager:
     threshold:
       controller_manager:
         periodicity:
+          window_size: {
+            type: int,
+            default_value: -1,
+            description: "The number of samples to consider for the mean. The diagnostics will be computed over the last `window_size` samples. -1 means that all samples will be considered.",
+          }
```

```suggestion
          window_size: {
            type: double,
            default_value: .inf,
            description: "The window size in seconds to consider for the mean. The diagnostics will be computed over the last `window_size` seconds. Default value infinity means that all data points will be considered.",
            validation: {
                gt<>: 0.0,
              }
          }
```

I believe the above validation works if you leave it with default value of infinity. I personally think it is more intuitive to the user to talk in seconds window rather than in the samples. What do you guys think?